morpheus/_lib/cmake/libmorpheus.cmake

```diff
@@ -46,6 +46,19 @@ target_link_libraries(cuda_objs
     matx::matx
 )
 
+set(DOCA_BUILD_FILES "")
+if(MORPHEUS_SUPPORT_DOCA)
+  set(DOCA_BUILD_FILES
+    src/doca/doca_context.cpp
+    src/doca/doca_rx_pipe.cpp
+    src/doca/doca_rx_queue.cpp
+    src/doca/doca_semaphore.cpp
+    src/doca/rte_context.cpp
+    src/stages/doca_source.cpp
+    src/stages/doca_source_kernels.cu
+  )
+endif()
+
```

Comment:
The DOCA and Morpheus components seem to be intertwined in this file. What if we built the DOCA pieces as a static library and then linked against it? That way, all of the DOCA specific settings and link options would be confined to a single library and all we would need to do is optionally link against it.

Reply:
Sounds like a good idea to me.